src/fides/api/service/connectors/saas/connector_registry_service.py

```diff
@@ -323,13 +323,14 @@ def create_connection_config_from_template_no_save(
     )
 
     data = {
-        "name": template_values.name,
-        "key": template_values.key,
+        "key": template_values.key if template_values.key else template_values.instance_key,
```

Comment:
This is where saas connectors get the instance key as a backup value now. 

The backend value used to be the name but that can't be relied on anymore since it's optional